orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt

```diff
@@ -421,6 +422,47 @@ class QueueIntegrationTest {
     }
   }
 
+  @Test
+  fun `terminal pipeline immediately cancels stages in other branches where tasks have long backoff times`() {
+    val pipeline = pipeline {
+      application = "spinnaker"
+      stage {
+        refId = "1"
+        type = "wait"
+        context = mapOf("waitTime" to 60)
+      }
+      stage {
+        refId = "2"
+        type = "dummy"
+        requisiteStageRefIds = emptyList()
+      }
+    }
+    repository.store(pipeline)
+
+    whenever(dummyTask.execute(argThat { refId == "2" })) doReturn TaskResult(TERMINAL)
+
+    context.runToCompletion(pipeline, runner::start, repository)
```

Comment:
This actually takes at least 60 seconds to return (and the test fails) without the `CancelStageHandler` portion of this PR. With it, this test doesn't appreciably add to `orca-queue-redis` test runtime.

Reply:
Sweet.